Assets/UI/worldinput_CQUI.lua

```diff
@@ -0,0 +1,559 @@
+    UI.SetInterfaceMode(InterfaceModeTypes.SELECTION);
+  end
+
+  return BASE_CQUI_OnUnitSelectionChanged( playerID, unitID, hexI, hexJ, hexK, isSelected, isEditable );
+end
+
+-- ===========================================================================
+function DefaultKeyDownHandler( uiKey:number )
+  print_debug("** Function Entry: DefaultKeyDownHandler (CQUI Hook).  uiKey: "..tostring(uiKey));
+  -- Note: This function always returns false, by design in Base game
+  BASE_CQUI_DefaultKeyDownHandler( uiKey ) ;
+
+  --CQUI Keybinds
+  local keyPanChanged :boolean = false;
+
+  if uiKey == Keys.VK_SHIFT then
+    print_debug("CQUI_isShiftDown = true");
+    CQUI_isShiftDown = true;
+  end
+
+  if uiKey == Keys.VK_ALT then
+    print_debug("CQUI_isAltDown = true");
+    CQUI_isAltDown = true;
+  end
+
+  if CQUI_hotkeyMode ~= CQUI_HOTKEYMODE_STANDARD then
+    if CQUI_hotkeyMode == CQUI_HOTKEYMODE_ENHANCED then
+      if (uiKey == Keys.W) then
+        keyPanChanged = true;
+        m_isUPpressed = true;
+      end
+
+      if (uiKey == Keys.D) then
+        keyPanChanged = true;
+        m_isRIGHTpressed = true;
+      end
+
+      if (uiKey == Keys.S) then
+        keyPanChanged = true;
+        m_isDOWNpressed = true;
+      end
+
+      if (uiKey == Keys.A) then
+        keyPanChanged = true;
+        m_isLEFTpressed = true;
+      end
+    end
+  end
+
+  if (keyPanChanged == true) then
+    -- Base game file uses m_edgePanX and m_edgePanY... but those values do not appear to ever change from 0, so just send 0.
+    ProcessPan(0, 0);
+    return true;
+  end
+
+  -- Allow other handlers of KeyDown events to capture input, if any exist after World Input
+  return false;
+end
+
+-- ===========================================================================
+function DefaultKeyUpHandler( uiKey:number )
+  print_debug("** Function Entry: DefaultKeyUpHandler (CQUI Hook).  uiKey: "..tostring(uiKey));
+
+  -- If set to Standard, just let the game do its thing
+  if CQUI_hotkeyMode == CQUI_HOTKEYMODE_STANDARD then
+    if uiKey == Keys.VK_SHIFT then
+      CQUI_isShiftDown = false;
+    end
+  
+    if uiKey == Keys.VK_ALT then
+      CQUI_isAltDown = false;
+    end
+
+    return BASE_CQUI_DefaultKeyUpHandler(uiKey);
+  end
+
+  local keyPanChanged  :boolean = false;
+  local cquiHandledKey :boolean = false;
+
+  local selectedUnit = UI.GetHeadSelectedUnit();
+  local unitType = nil;
+  local formationClass = nil;
+  if (selectedUnit) then
+    unitType = GameInfo.Units[selectedUnit:GetUnitType()].UnitType;
+    formationClass = GameInfo.Units[selectedUnit:GetUnitType()].FormationClass;
+  end
+
+  --CQUI Keybinds
+  if CQUI_hotkeyMode == CQUI_HOTKEYMODE_ENHANCED then
+    if (uiKey == Keys.W) then
+      m_isUPpressed = false;
+      keyPanChanged = true;
+    end
+
+    if (uiKey == Keys.D) then
+      m_isRIGHTpressed = false;
+      keyPanChanged = true;
+    end
+
+    if (uiKey == Keys.S) then
```

Comment:
Couldn't this overlap with the Alt+S handling near end of this function. I'm thinking it might result in rebase and map move if you press Alt+S

Reply:
hmm I think you're right - good catch - if holding down alt and letting go of 'S', it may catch this here.  Will add the if altdown==false clause